NOTICE-binary

```diff
@@ -42,6 +42,42 @@ Copyright 2005-2024 The Apache Software Foundation
 Apache Commons Lang
 Copyright 2001-2024 The Apache Software Foundation
 
+Apache Commons Pool
```

Comment:
IIUC, we don't need to mention the ASF project here?

```
- binary distribution NOTICE should content the NOTICE from ALv2
project (non ASF), for instance Jackson, AWS SDK, Netty, etc. That
would be great to double check that for next releases.
```

Reply:
And we can remove the ASF project listed in the `NOTICE` file

```
- NOTICE is OK. Nit: the NOTICE can be lighter: no need to mention the
ASF projects here (Iceberg, Paimon, Flink)
```

Reply:
Yes, it is fine to exclude the ASF project, but I still prefer to include it here like many other projects did:

- https://github.com/apache/gravitino/blob/main/NOTICE.bin
- https://github.com/apache/spark/blob/master/NOTICE-binary

Reply:
I'm fine with either way, but as the [asf website](https://infra.apache.org/licensing-howto.html#mod-notice) said, so maybe removing the ASF project here is better.

```
It is important to keep NOTICE as brief and simple as possible, as each addition places a burden on downstream consumers.

**Do not** add anything to NOTICE which is not legally required.
```

Reply:
Make sense to me, I will remove the ASF projects.